src/lib.rs

```diff
@@ -640,6 +645,113 @@ impl fmt::Debug for StructuredExtendedInformation {
     }
 }
 
+#[derive(Copy,Clone)]
+pub struct ProcessorExtendedState {
+    eax: u32,
+    ebx: u32,
+    ecx: u32,
+}
+
+impl ProcessorExtendedState {
+    fn new() -> ProcessorExtendedState {
+        let (a, b, c, _) = cpuid(RequestType::ProcessorExtendedState);
+        ProcessorExtendedState { eax: a, ebx: b, ecx: c }
+    }
+
+    bit!(eax, {
+        0 => x87_state,
+        1 => sse_state,
+        2 => avx_state,
+        // 3-4 mpx_state
+        // 5-7 avx_512_state
+        8 => ia32_xss,
+        9 => pkru_state
+        // 10-31 - reserved
+    });
+
+    pub fn mpx_state(self) -> u32 {
+        bits_of(self.eax, 3, 4)
+    }
+
+    pub fn avx_512_state(self) -> u32 {
+        bits_of(self.eax, 5, 7)
+    }
+
+    pub fn maximum_bytes_for_enabled_features(self) -> u32 {
+        self.ebx
+    }
+
+    pub fn maximum_bytes_for_supported_features(self) -> u32 {
+        self.ecx
+    }
+}
+
+impl fmt::Debug for ProcessorExtendedState {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        dump!(self, f, "ProcessorExtendedState", {
+            x87_state,
+            sse_state,
+            avx_state,
+            mpx_state,
+            avx_512_state,
+            ia32_xss,
+            pkru_state,
+
+            maximum_bytes_for_enabled_features,
+
+            maximum_bytes_for_supported_features
+        })
+    }
+}
+
+#[derive(Copy,Clone)]
+pub struct ProcessorExtendedStateSecondary {
+    eax: u32,
+    ebx: u32,
+    ecx: u32,
+}
```

Comment:
`edx` is missing here: 

> Bits 31 - 00: Reports the supported bits of the upper 32 bits of the IA32_XSS MSR. IA32_XSS[n+32] can be set to 1 only if EDX[n] is 1.

---

Should also add a function returning a `u32` below to access this.

Reply:
Walk me through the usefulness of this. The next line of the manual says:

> Bits 31 - 00: Reserved.

I'm not immediately seeing the value in returning 32bits of reserved values.

Reply:
> The next line of the manual says:

I did not read that line 😆